src/client/websocket/handlers/CHANNEL_PINS_UPDATE.js

```diff
@@ -6,7 +6,7 @@ module.exports = (client, { d: data }) => {
   const channel = client.channels.get(data.channel_id);
   const time = new Date(data.last_pin_timestamp);
 
-  if (channel && time) {
+  if (channel && !Number.isNaN(time.getTime())) {
```

Comment:
If I'm reading the code right, this means it won't emit the event once the last pinned message was unpinned, which doesn't seem intended.